test/issue-25-default-values.cpp

```diff
@@ -28,6 +28,25 @@ static const json person_schema = R"(
                     "default": "Abbey Road"
                 }
             }
+        },
+        "work address": {
+            "type": "object",
+            "default": null,
+            "properties": {
+                "street": {
+                    "type": "string",
+                    "default": "Abbey Road"
+                }
+            }
+        },
```

Comment:
This is interesting! The parent object of `street` has default = null, but street has a non-null default.

Why do we assume that a parent default is more important than a child one? Same as below: no parent-default means child defaults are ignored.

This is an issue which needs to be looked into.

Reply:
https://github.com/pboettch/json-schema-validator/issues/124

Reply:
Not sure it is a issue. Let me explain a bit our context:

My journey started when I was willing to generate default out of schema (I have seen, even if it is pure annotation, that need seem common). I looked to this library we were already using and discovered that it generate a patch with default value as long as the parent object exist. So to generate default, I do:
 - remove all required inside the schema
 - start with empty object
 - validate with schema
 - apply patch
 - repeat until patch is a empty list

And it work fine. It is a bit strange and a method on the library that do this job would be great but at the end it work.

Of course, another way would be to repeat all member on the default object up to root. But repeating over and over values doesn't sound good to me.